src/Windows/R/Editor/Impl/Settings/REditorSettings.cs

```diff
@@ -24,15 +26,28 @@ public sealed class REditorSettings: EditorSettings, IWritableREditorSettings, I
 
         private RFormatOptions _formatOptions = new RFormatOptions();
 
-        public REditorSettings(ICoreShell coreShell): base(coreShell, RContentTypeDefinition.ContentType) { }
-        public REditorSettings(IEditorSettingsStorage storage) : base(storage) { }
+        public REditorSettings(ICoreShell coreShell) : base(coreShell, RContentTypeDefinition.ContentType) {
+            CreateLintOptions();
+        }
+
+        public REditorSettings(IEditorSettingsStorage storage) : base(storage) {
+            CreateLintOptions();
+        }
 
         public override void ResetSettings() {
             _formatOptions = new RFormatOptions();
-            LintOptions = new LintOptions();
+            CreateLintOptions();
             base.ResetSettings();
         }
 
+        private void CreateLintOptions() {
+            LintOptions = new LintOptions();
+            LintOptions.PropertyChanged += _OnLintOptionsPropertyChanged;
```

Comment:
Then `LintOptions` can follow the same pattern as `REditorSettings`. `RLintOptionsDialog` already follows `RLintOptionsDialog` pattern, so `LintOptions` can use `Storage` and `WritableStorage`, something like this:
```cs
    public sealed class LintOptions {
        public const string EnabledKey = nameof(LintOptions) + ":" + nameof(Enabled);

        private readonly IEditorSettingsStorage _storage;
        private readonly IWritableEditorSettingsStorage _writableStorage;

        public class LintOptions (IEditorSettingsStorage storage) {
            _storage = storage;
            _writableStorage = storage as IWritableEditorSettingsStorage;
        }

        public bool Enabled {
            get => Storage.Get(EnabledKey, true);
            set => WritableStorage?.Set(EnabledKey, value);
        }
```